### ISSUE TITLE
blank image when I download a chart

### ISSUE THREAD

**Issue by AllenDowney** [owid/owid-grapher#4332]
I'm getting a blank image when I try to download the PNG version of the graph on this page:

https://ourworldindata.org/grapher/countries-democracies-autocracies-row

The preview is blank, and the downloaded image if blank. The SVG version is fine.

The problem seems to be Firefox only -- it works in Chromium.

## Screenshots

![Image](https://github.com/user-attachments/assets/c0b0f43f-bbe8-4ae3-afd0-6935e8515338)

## Environment

For desktop, provide:

- OS: Linux
- Browser Firefox 133.0.3
-

**Comment by ikesau** [comment]
Hey @AllenDowney thanks for the report! 🙂 

This might be a linux thing as it's working for me on my MacOS Firefox v134

Can you see if there's any logging / errors in your console output?

**Comment by AllenDowney** [comment]
Yes!

```
The resource at “https://github.com/_graphql?body=%7B%22query%22%3A%22d569a3ee4ed5313a5d887b2066bf2965%22%2C%22variables%22%3A%7B%22markAsRead%22%3Atrue%2C%22number%22%3A4332%2C%22owner%22%3A%22owid%22%2C%22repo%22%3A%22owid-grapher%22%2C%22useNewTimeline%22%3Atrue%7D%7D” preloaded with link preload was not used within a few seconds. Make sure all attributes of the preload tag are set correctly.
```

**Comment by ikesau** [comment]
That's the message that logs on ourworldindata.org when you get the blank image?

I don't think we interact with github's graphql API at all so maybe that's an extension of yours, or a log from a different tab?

**Comment by AllenDowney** [comment]
Yeah, sorry -- I should have noticed that. So no, I don't think I'm getting an error -- as far as the browser is concerned, it seems like a valid PNG, and after I download it, I can view it without errors, but it's blank:
![Image](https://github.com/user-attachments/assets/b764181b-d21b-45be-8181-39bbbf9d2736)

**Comment by ikesau** [comment]
Okay and just to be 100% crystal clear - if you do the following, you don't get any visible error logging?

https://github.com/user-attachments/assets/5e1eb0e6-1c7d-4f43-be11-5c3763670b65

Sorry for the persistance, but unless we get some obvious breadcrumbs here it might be a bit tricky for us to debug this as I don't think anyone on the team is running a linux machine atm, and it's also not likely to affect many of our users 😅 

In any case, if it helps, we have a cloudflare worker setup that allows you to add `.png` to the end of any grapher / explorer slug to get a PNG version of the chart

e.g.

https://ourworldindata.org/grapher/countries-democracies-autocracies-row.png

or

https://ourworldindata.org/explorers/co2.png

and you can add [query params](https://en.wikipedia.org/wiki/Query_string) afterwards for timeline/entity selection, etc

https://ourworldindata.org/grapher/countries-democracies-autocracies-row.png?time=1959..latest&country=OWID_EUR~OWID_OCE

You can also add `&imWidth=1000` to set the width of the generated image.

This should work regardless of your operating system! 🙂

**Comment by AllenDowney** [comment]
There is one error in the console, but I don't know if it's relevant

```
Loading failed for the <script> with source “https://static.cloudflareinsights.com/beacon.min.js”.
```

**Comment by danyx23** [comment]
Thanks for your bug report @AllenDowney! We're planning to switch png download from the current, client side generation to the png server download that Ike has outlined above. Because of this roadmap, we'll not investigate this issue further. I hope the PNG download option that Ike described will work for you until we switch our download button in the future!

**Comment by AllenDowney** [comment]
Sounds good. Thanks for the update.
